lib/js_of_ocaml/typed_array.ml

```diff
@@ -49,47 +51,61 @@ class type ['a, 'b] typedArray = object
 
-  method slice : int -> int -> ('a, 'b) typedArray t meth
+  method slice : int -> int -> ('a, 'b, 'c) typedArray t meth
 
-  method slice_toEnd : int -> ('a, 'b) typedArray t meth
+  method slice_toEnd : int -> ('a, 'b, 'c) typedArray t meth
 
   (* This fake method is needed for typing purposes.
      Without it, ['b] would not be constrained. *)
-  method _content_type_ : 'b optdef readonly_prop
+  method _content_type_ : ('b * 'c) optdef readonly_prop
 end
 
-type int8Array = (int, Bigarray.int8_signed_elt) typedArray
+type int8Array = (int, int, Bigarray.int8_signed_elt) typedArray
+
+type uint8Array = (int, int, Bigarray.int8_unsigned_elt) typedArray
 
-type uint8Array = (int, Bigarray.int8_unsigned_elt) typedArray
+type int16Array = (int, int, Bigarray.int16_signed_elt) typedArray
 
-type int16Array = (int, Bigarray.int16_signed_elt) typedArray
+type uint16Array = (int, int, Bigarray.int16_unsigned_elt) typedArray
 
-type uint16Array = (int, Bigarray.int16_unsigned_elt) typedArray
+type int32Array = (int32, Int32.t, Bigarray.int32_elt) typedArray
 
-type int32Array = (int32, Bigarray.int32_elt) typedArray
+type uint32Array = (uint32, Int32.t, Bigarray.int32_elt) typedArray
 
-type uint32Array = (int32, Bigarray.int32_elt) typedArray
+type float32Array = (number_t, float, Bigarray.float32_elt) typedArray
 
-type float32Array = (float, Bigarray.float32_elt) typedArray
+type float64Array = (number_t, float, Bigarray.float64_elt) typedArray
 
-type float64Array = (float, Bigarray.float64_elt) typedArray
+type ('bigarray, 'typed_array, 'elt) type' =
+  | Char : (int, char, Bigarray.int8_unsigned_elt) type'
+  | Int8_signed : (int, int, Bigarray.int8_signed_elt) type'
+  | Int8_unsigned : (int, int, Bigarray.int8_unsigned_elt) type'
+  | Int16_signed : (int, int, Bigarray.int16_signed_elt) type'
+  | Int16_unsigned : (int, int, Bigarray.int16_unsigned_elt) type'
+  | Int32_signed : (int32, Int32.t, Bigarray.int32_elt) type'
+  | Int32_unsigned : (uint32, Int32.t, Bigarray.int32_elt) type'
+  | Float32 : (number_t, float, Bigarray.float32_elt) type'
+  | Float64 : (number_t, float, Bigarray.float64_elt) type'
 
-external kind : ('a, 'b) typedArray t -> ('a, 'b) Bigarray.kind
+external kind :
+  ('typed_array, 'bigarray, 'elt) typedArray t -> ('bigarray, 'elt) Bigarray.kind
   = "caml_ba_kind_of_typed_array"
 
-external from_genarray :
-  ('a, 'b, Bigarray.c_layout) Bigarray.Genarray.t -> ('a, 'b) typedArray t
-  = "caml_ba_to_typed_array"
+external from_genarray_impl :
+     ('bigarray, 'elt, Bigarray.c_layout) Bigarray.Genarray.t
+  -> ('typed_array, 'bigarray, 'elt) typedArray t = "caml_ba_to_typed_array"
 
 external to_genarray :
-  ('a, 'b) typedArray t -> ('a, 'b, Bigarray.c_layout) Bigarray.Genarray.t
-  = "caml_ba_from_typed_array"
+     ('typed_array, 'bigarray, 'elt) typedArray t
+  -> ('bigarray, 'elt, Bigarray.c_layout) Bigarray.Genarray.t = "caml_ba_from_typed_array"
+
+let from_genarray _ a = from_genarray_impl a
```

Comment:
adding a type constraint to the `_` would help !

Reply:
Done